ciborium/src/ser/mod.rs

```diff
@@ -7,24 +7,57 @@ mod error;
 pub use error::Error;
 
 use alloc::string::ToString;
-
 use ciborium_io::Write;
 use ciborium_ll::*;
 use serde::{ser, Serialize as _};
 
-struct Serializer<W>(Encoder<W>);
+/// A serializer for CBOR.
+pub struct Serializer<W> {
+    encoder: Encoder<W>,
+
+    /// Whether to canonically sort map keys in output according to [RFC 8949]'s deterministic
+    /// encoding spec.
+    ///
+    /// [RFC 8949]: https://www.rfc-editor.org/rfc/rfc8949.html#name-deterministically-encoded-c
+    #[cfg(feature = "std")]
+    canonical: bool,
+}
+
+impl<W: Write> Serializer<W> {
+    /// Create a new CBOR serializer.
+    ///
+    /// `canonical` determines whether to canonically sort map keys in output according to
+    /// [RFC 8949]'s deterministic encoding spec. Requires the `std` feature to sort keys.
+    ///
+    /// [RFC 8949]: https://www.rfc-editor.org/rfc/rfc8949.html#name-deterministically-encoded-c
+    #[cfg(feature = "std")]
+    pub fn new(encoder: impl Into<Encoder<W>>, canonical: bool) -> Self {
+        Self {
+            encoder: encoder.into(),
+            canonical
+        }
+    }
+}
 
 impl<W: Write> From<W> for Serializer<W> {
     #[inline]
     fn from(writer: W) -> Self {
-        Self(writer.into())
+        Self {
+            encoder: writer.into(),
+            #[cfg(feature = "std")]
```

Comment:
The `#[cfg(feature = "std")]` gating here is perhaps  unnecessary, I'd need to check.